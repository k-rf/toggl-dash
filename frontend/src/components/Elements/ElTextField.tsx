import { TextField, TextFieldProps } from "@mui/material";
import { UseFormRegisterReturn } from "react-hook-form";

type Props = Pick<
  TextFieldProps,
  | "label"
  | "size"
  | "fullWidth"
  | "error"
  | "helperText"
  | "onKeyDown"
  | "placeholder"
  | "type"
  | "InputProps"
  | "id"
> & { registration: Partial<UseFormRegisterReturn> };

export const ElTextField = (props: Props) => {
  const { registration, ...others } = props;

  return <TextField {...others} {...registration} />;
};
