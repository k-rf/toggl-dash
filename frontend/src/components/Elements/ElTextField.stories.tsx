import type { Meta, StoryObj } from "@storybook/react";

import { ElTextField, ElTextFieldProps } from "./ElTextField";

export default { component: ElTextField } as Meta<ElTextFieldProps>;

type Story = StoryObj<ElTextFieldProps>;

export const Default: Story = {};
