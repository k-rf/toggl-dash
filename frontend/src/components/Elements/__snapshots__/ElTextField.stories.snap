// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Elements/ElTextField > Default 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
      disabled={false}
      id=":r0:"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
    />
    <fieldset
      aria-hidden={true}
      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="css-ihdtdm"
      >
        <span
          className="notranslate"
        >
          â€‹
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;
